cmd/skaffold/app/cmd/deploy.go

```diff
@@ -17,24 +17,54 @@ limitations under the License.
 package cmd
 
 import (
+	"context"
 	"io"
 
+	"github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/flags"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build"
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
+var (
+	buildOutputFile flags.BuildOutputFileFlag
+	preBuiltImages  flags.Images
+)
+
 // NewCmdDeploy describes the CLI command to deploy artifacts.
 func NewCmdDeploy(out io.Writer) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "deploy",
 		Short: "Deploys the artifacts",
 		Args:  cobra.NoArgs,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// Same actions as `skaffold run`, but with pre-built images.
-			return run(out)
+			return runDeploy(out)
 		},
 	}
 	AddRunDevFlags(cmd)
 	AddRunDeployFlags(cmd)
-	cmd.Flags().StringSliceVar(&opts.PreBuiltImages, "images", nil, "A list of pre-built images to deploy")
+	cmd.Flags().VarP(&preBuiltImages, "images", "i", "A list of pre-built images to deploy")
+	cmd.Flags().VarP(&buildOutputFile, "build-artifacts", "a", "Output of `skaffold build --quiet {{json .}} > build.out`")
```

Comment:
probably shouldn't use `build.out` in the description here, since that output could actually have been any arbitrary file

Reply:
added a better help message